primitives/src/crh/poseidon/mod.rs

```diff
@@ -347,4 +347,36 @@ mod test {
         assert_eq!(output, expected_output, "Outputs do not match for BN382Fq");
     }
 
+    use algebra::{
+        fields::tweedle::{
+            fr::Fr as tweedleFr, fq::Fq as tweedleFq,
+        },
+        biginteger::BigInteger256,
+    };
+    use crate::crh::parameters::tweedle::*;
+
+    #[test]
+    fn test_poseidon_hash_tweedle_fr() {
+        let expected_output = tweedleFr::new(BigInteger256([10853721058648678319, 16685221779982166148, 7657542961996224896, 317411048550701368]));
```

Comment:
Again: where are the test vectors from?

Reply:
Extensive test vectors can be found in the branch [UpdatedPoseidonParameters](https://github.com/HorizenOfficial/ginger-lib/tree/UpdatedPoseidonParameters).